compiler/rustc_codegen_ssa/src/mir/analyze.rs

```diff
@@ -17,11 +17,14 @@ use rustc_target::abi::LayoutOf;
 
 pub fn non_ssa_locals<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
     fx: &FunctionCx<'a, 'tcx, Bx>,
+    reverse_postorder: &mut traversal::ReversePostorder<'tcx, 'tcx>,
 ) -> BitSet<mir::Local> {
     let mir = fx.mir;
     let mut analyzer = LocalAnalyzer::new(fx);
 
-    analyzer.visit_body(&mir);
+    for (bb, data) in reverse_postorder.reset() {
```

Comment:
Callees having to reset the iterator seems quite error prone. Callers having to do so seems as bad, though.

I would perhaps reach for the ability to cheaply clone the `ReversePostorder` iterator and pass in a “fresh”-looking one into each user that wants to reuse the post-order iteration. :shrug:

Reply:
I would rather drop changes related to the reuse of reverse post-order traversal, unless they have easily measurable positive perf impact. Lets, run perf again without them.

Reply:
The perf impacts looks negligible either way, so limited changes to a use of a different traversal. Thanks, for looking.